Sources/Swift/Protocol/Codable/SentryCodable.swift

```diff
@@ -1,6 +1,18 @@
 @_implementationOnly import _SentryPrivate
 import Foundation
 
+func encodeToJSONData<T: Encodable>(data: T) throws -> Data {
+    let jsonEncoder = JSONEncoder()
+    jsonEncoder.dateEncodingStrategy = .custom({ (date, encoder) in
+        let formatter = sentryGetIso8601FormatterWithMillisecondPrecision()
+        let dateString = formatter.string(from: date)
```

Comment:
`h`: I would prefer using a double value for the date cause it allows nanoseconds precision. This date formatter only allows milliseconds. We're encoding here so we should ditch the milliseconds precision here. We only use it in decode because some properties still use milliseconds precision.